docs/essentials/parameterized-promotions.md

```diff
@@ -0,0 +1,187 @@
+          commands:
+            # Use parameter values inside a job
+            - echo $ENVIRONMENT
+            - echo $RELEASE
+      secrets:
+        # Use parameter values when attaching a secret
+        - name: 'creds-for-${{parameters.ENVIRONMENT}}'
+```
+**Note:** For this example pipeline to work, secrets named `creds-for-Production` and `creds-for-Stage` have to exist in the organization. 
+
+## Setting the parameter values
+Now that you have configured both the initial pipeline and the promotion pipeline, we can start using parameterized promotions. 
+
+Values to the parameters are assigned every time a promotion is triggered. This can be done in the following ways:
+
+- In the UI form if promotion is triggered manually
+- Through the API call 
+- Through `default_value` in combination with `auto_promote`
+
+### Setting the values through the UI
+In the Semaphore web interface, you will see a deployment button. 
+
+When clicked, a UI form will be auto-generated based on the configuration in the initial pipeline, prompting you for parameter values before starting the promotion. 
+
+Here's what it looks like for the `.yml` code from our example:
+
+<img style="box-shadow: 0px 0px 5px #ccc" src="/essentials/img/parameterized-promotions/ui-form.png" alt="Tests Tab on Workflow Page">
+
+### Setting the values through the API
+Parameters can also be passed to the promotion pipeline when the promotions are triggered through Semaphore API. 
+
+The following is an example of a curl call that includes parameters:
+```
+curl -H "Authorization: Token {api_token}" \
+     -d "name={promotion_name}&pipeline_id={parent_pipeline_id}&{param_1_name}={param_1_value}&{param_2_name}={param_2_value}" \
+     -X POST "https://{org_name}.semaphoreci.com/api/v1alpha/promotions"
+```
+
+### Setting the values through the default value
+If you are using the [auto_promote](https://docs.semaphoreci.com/reference/pipeline-yaml-reference/#auto_promote) to trigger the promotion, you can still set the value for the promotion parameters. 
+
+This can be done by utilizing the `default_value` option:
+``` yaml
+# .semaphore/semaphore.yml
+version: v1.0
+name: Parameterized Promotion
+agent:
+  machine:
+    type: e1-standard-2
+    os_image: ubuntu2004
+blocks:
+  - name: Promotions
+    task:
+      jobs:
+        - name: Everything
+          commands:
+            - echo 'Running tests'
+promotions:
+  - name: Deploy
+    pipeline_file: deploy.yml
+    # Define promotion parameters
+    auto_promote:
+      when: pull_request =~ '.*' AND result = 'passed'
+    parameters:
+      env_vars:
+        - required: true
+          options:
+            - Stage
+            - Production
+          default_value: Stage
+          description: Where to deploy?
+          name: ENVIRONMENT
+       
+        - required: true
+          options: []
+          description: Release name?
+          name: RELEASE
+          default_value: $SEMAPHORE_GIT_PR_NAME
```

Comment:
This will not work, the value will be the actual string `"$SEMAPHORE_GIT_PR_NAME"` and not the value of the environment variable with that same name.